### PR TITLE
set default quota in contrail

### PR DESCRIPTION
**Pull request by sajuptpm**
Set default quota for floating_ip, logical_router,
security_group, security_group_rule, subnet,
virtual_machine_interface and virtual_network.

### CODE REVIEW

File: hiera/data/common.yaml [hunk 1]
```diff
@@ -654,6 +654,13 @@ rjil::contrail::server::enable_analytics: false
 contrail::interface: "%{hiera('public_interface')}"
 contrail::vrouter::vrouter_physical_interface: "%{hiera('contrail::interface')}"
 
+contrail::config::quota_floating_ip: 10
+contrail::config::quota_logical_router: 20
+contrail::config::quota_security_group: 50
+contrail::config::quota_security_group_rule: 50
+contrail::config::quota_subnet: 20
+contrail::config::quota_virtual_machine_interface: 30
+contrail::config::quota_virtual_network: 8
```
Comment (bodepd) [inline hiera/data/common.yaml]: are these values just mirroring the defaults? In that case, I'd rather not have them specified in hiera since our hiera common file is already a bit unwieldy.

Comment (sajuptpm) [inline hiera/data/common.yaml]: no, changed defaults in contrail::config to -1.

**Comment by sajuptpm** [comment]
please test this

**Comment by sajuptpm** [comment]
please test this

**Comment by sajuptpm** [comment]
please test this

File: .fixtures.yml [hunk 2]
```diff
@@ -32,7 +32,8 @@ fixtures:
       repo: https://github.com/saz/puppet-ssh
       ref: v2.3.6
     contrail:
-      repo: https://github.com/jiocloud/jiocloud-contrail
```
Comment (bodepd) [inline .fixtures.yml]: this has been merged.

File: Puppetfile [hunk 3]
```diff
@@ -142,8 +142,8 @@ mod 'stephenrjohnson/puppet',
   :ref => '1.0.0'
 
 mod 'jiocloud/contrail',
-  :git => "#{base_url}/jiocloud/jiocloud-contrail",
```
Comment (bodepd) [inline Puppetfile]: this has been merged.

**Comment by bodepd** [comment]
I'd love to see this rebased as two commits, have the revisions updated, then tests run one more time

File: hiera/data/common.yaml [hunk 4]
```diff
@@ -654,6 +654,13 @@ rjil::contrail::server::enable_analytics: false
 contrail::interface: "%{hiera('public_interface')}"
 contrail::vrouter::vrouter_physical_interface: "%{hiera('contrail::interface')}"
 
+contrail::config::quota_floating_ip: 5
```
Comment (bodepd) [inline hiera/data/common.yaml]: is there a ticket that tracks the status of product management decisions related to default quotas?

**Comment by bodepd** [comment]
one more comment about this patch, the commit message isn't really descriptive enough. It mentions what the patch does, but not why. As in, why as the values being set, why were these defaults selected. The intention of the commit message is that is should be descriptive enough to provide enough context to understand everything you need to know about the changes.

File: hiera/data/common.yaml [hunk 5]
```diff
@@ -654,6 +654,13 @@ rjil::contrail::server::enable_analytics: false
...
+contrail::config::quota_logical_router: 10
+contrail::config::quota_security_group: 100
+contrail::config::quota_security_group_rule: 50
+contrail::config::quota_subnet: 20
+contrail::config::quota_virtual_machine_interface: 100
```
Comment (hkumarmk) [inline hiera/data/common.yaml]: I dont think we should restrict vm interfaces, this should be based on nova instance quota and should use default quota of unlimited vm interfaces.

File: hiera/data/env/at.yaml [hunk 6]
```diff
@@ -7,8 +7,8 @@ rjil::ceph::osd::osd_journal_size: 2
 # Only Qemu is supported on virtual testing environment
 nova::compute::libvirt::libvirt_virt_type: qemu
 
-rjil::system::accounts::active_users: [soren,bodepd,hkumar,jenkins,consul,pandeyop,jaspreet,vivek,ahmad,vaidy,himanshu,rohit,sanjayu,alokjani,amar,ynshenoy,abhishekl,mayankkapoor,soumit,anshup,varunarya,prashant,punituee]
-rjil::system::accounts::sudo_users: [soren,bodepd,hkumar,jenkins,consul,pandeyop,jaspreet,vivek,ahmad,vaidy,himanshu,rohit,sanjayu,alokjani,amar,ynshenoy,abhishekl,mayankkapoor,soumit,anshup,varunarya,prashant,punituee]
+rjil::system::accounts::active_users: [soren,bodepd,hkumar,jenkins,consul,pandeyop,jaspreet,vivek,ahmad,vaidy,himanshu,rohit,sanjayu,alokjani,amar,ynshenoy,abhishekl,mayankkapoor,soumit,anshup,varunarya,prashant,punituee,saju]
```
Comment (hkumarmk) [inline hiera/data/env/at.yaml]: I would suggest to add this change to separate PR (this change is not related to what this patch meant for.

File: hiera/data/secrets/common.yaml [hunk 7]
```diff
@@ -21,6 +21,7 @@ rjil::system::accounts::local_users:
...
   punituee: {realname: 'Puneet Goyal', sshkeys: "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDDGLt9eU0RbNjkDnxZMQNE+YxbQ4nez5H8vnKKkN9fxbHAZT4KS8MV9Qlvi2pLbt7G2h7xiARQrbNCF4aqdeOaRVstSW6jWClTDUwL6U9S2IC8kdYqUzZ8R9RdxcgH5EtvLGPxaIycIj7UAMdlTZfbx3jwJ+FhSW1WmoPe3ditbMNnazf78iYVJyAFta2I68Xg1+2MQi68EPpcvgvm+tfUiilV5Tol2tbmpjkprWPMhCNMp6MUe/mptXVoJnwO1Ey/kAkwZ1XYhLy2MviTCZvfktnquhf0ANdE9RDP/aLMMLU+CtX+djuKdGIlDivJVhL40lFJUOzw4yz8SpudkTEH puneetgoyal728@gmail.com", password: '*'}
+  saju: {realname: 'Saju Madhavan', sshkeys: "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDbVis4+ocEBacnMeNwpG0NCf41wPgETsBA8GIQQh6V/B5CBDMvcdr1c7PPtjvWN9Cmf6Bh1YMATUPeceFVUuHPhdihDG7rwKCalUKd0ims/fCdZUqXgkPsZG4KgUTQcqg/bbibJeWZ4f5GxVhwVmCacdZ9yJqyNwYJN6YnDV/mKUovCMvk/vs5JK4tWp6czQhjGoxg59b70IKOirZlhALNGRHfnl9iVewB37S8S5Ehm3J5VxfA246K2uqyWb1OD5Yk9t+aKoWeFjUspG+XOuUx7cFdHdno6nXpiMxPuO0hPi+xBG9BeqghkqXu5VYvei6p4f98zQl67RLLvNdg3UJd saju@saju-Inspiron-5521", password: '*'}
```
Comment (hkumarmk) [inline hiera/data/secrets/common.yaml]: I would suggest this change to separate PR.

**Comment by sorenh** [comment]
The actual values for production can/should be overridden by the internal hiera anyway, so I don't think we need to worry too much about the specific values.

With that in mind, does anyone have any reservations about this patch?

**Comment by hkumarmk** [comment]
Other than this I am good to get it merged. so merging...

**Comment by bodepd** [comment]
my only reservation was the commit message, as more folks come on board, I would like things to be a little more descriptive in terms of:

````
what changes are being made
why were the changes made
what are the results
````

Otherwise, it's going to be really difficult for us to understand how things got into the state that they are in.

That being said, this is probably a higher level conversation that needs to be had around process.
